zephyr/src/utils.c

```diff
@@ -660,6 +660,20 @@ int find_interface_ip(char *ipaddr, int ipaddr_len, char *name) {
 }
 
 int get_mac_address(char *buffer, int size, char *interface) {
+
+    STRUCT_SECTION_FOREACH(net_if, iface)
```

Comment:
Again these are internal details which can easily throw away iterable sections and move to another storage implementation and break this code, please use the public API `net_if_foreach`  + `net_if_is_wifi`.